plugins/osdn/netid/migration/migration.go

```diff
@@ -0,0 +1,89 @@
+package migration
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/storage"
+
+	"github.com/openshift/openshift-sdn/pkg/netid"
+	"github.com/openshift/origin/pkg/sdn/api"
+)
+
+const (
+	etcdNetNamespacePrefix string = "/registry/sdnnetnamespaces"
+)
+
+type Migrate struct {
+	etcdHelper storage.Interface
+	client     kclient.NamespaceInterface
+}
+
+func NewMigrate(etcdHelper storage.Interface, client kclient.NamespaceInterface) *Migrate {
+	return &Migrate{
+		etcdHelper: etcdHelper,
+		client:     client,
+	}
+}
+
+func (m *Migrate) Run() error {
+	var netnsList api.NetNamespaceList
+
+	err := m.etcdHelper.List(context.TODO(), etcdNetNamespacePrefix, "", storage.Everything, &netnsList)
+	if err != nil {
+		return fmt.Errorf("List network namespaces failed: %v", err)
+	}
+
+	for _, netns := range netnsList.Items {
+		// Get corresponding namespace for the NetNamespace object
+		ns, err := m.client.Get(netns.NetName)
+		if errors.IsNotFound(err) {
+			m.deleteNetNamespace(&netns)
+			continue
+		}
+		if err != nil {
+			return fmt.Errorf("Get namespace %q failed: %v", netns.NetName, err)
```

Comment:
all of the errors in this function need to explain to the admin what is going on and what is actionable.  None of these are good enough to debug a migration in progress.